src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4259,12 +4265,112 @@ private static BoundExpression SkipReferenceConversions(BoundExpression possibly
             }
         }
 
-        public override BoundNode? VisitConditionalAccess(BoundConditionalAccess node)
+        // PROTOTYPE(improved-definite-assignment):
+        // this is not used outside 'VisitPossibleConditionalAccess' yet, but is expected to be used
+        // when refactoring 'VisitBinaryOperatorChildren'
+        private bool TryVisitConditionalAccess(BoundExpression node, out PossiblyConditionalState stateWhenNotNull)
+        {
+            var (conversion, access) = node switch
+            {
+                BoundConditionalAccess ca => (null, ca),
+                BoundConversion { Conversion: Conversion innerConversion, Operand: BoundConditionalAccess ca } conv when isAcceptableConversion(ca, innerConversion) => (conv, ca),
+                _ => default
+            };
+
+            if (access is object)
+            {
+                EnterRegionIfNeeded(access);
```

Comment:
Do the `Enter/LeaveRegionIfNeeded` calls matter?

Reply:
When doing the definite assignment side of this feature, I had to add these calls in the analogous position over there, because the usage of `VisitConditionalAccess(node, out stateWhenNotNull)` caused us to side-step the regular machinery for managing region analysis in `AbstractFlowPass.VisitAlways`. When this was missing in definite assignment, "extract method" and similar feature tests were broken.

I do not actually know what dependent features could be broken by the absence of this in NullableWalker, or how to test it in a more direct manner, but it seems like something that we should have anyway.

Reply:
Other nullable analysis methods don't have this, except that we set a value once in `Scan` (not exactly sure what that achieves either). I think we should remove it unless a scenario warrants it.

---
In reply to: [620688053](https://github.com/dotnet/roslyn/pull/52646#discussion_r620688053) [](ancestors = 620688053)

Reply:
Done